### PR TITLE
Machine upgrade examine verb

### PR DESCRIPTION
**Pull request by 0x6273**
~~Ignore the commits before a62f66d9a47fefefb3f11efc67ec14b670b6e609, they are unrelated to this change and will go away when #11972 gets merged and I rebase this on master.~~ **It's rebased now.**

Known issues:
- [x] ~~Needs loc strings.~~
- [x] ~~Only seed extractor and UpgradePowerDrawComponent implemented.~~
- [x] For some reason there is a blank line at the end of the examine tooltip.
    - I think this is due to a bug in RichTextEntry UI code, where `\n` is sometimes treated as a line feed rather than line break.
    - I added a workaround in 9591461c120f2492ac14c3187e5025fe0d2e84fd. It's ugly and should be removed when https://github.com/space-wizards/RobustToolbox/issues/3371 is fixed.
- [x] ~~Should probably add xml doc comments on the public methods.~~

**Screenshots**
image

**Changelog**
:cl:
- add: Upgradeable machines can now be examined to show the effects that the upgrades have.

### CODE REVIEW

File: Content.Server/Construction/ConstructionSystem.Machine.cs [hunk 1]
```diff
@@ -25,6 +31,31 @@ private void OnMachineMapInit(EntityUid uid, MachineComponent component, MapInit
         RefreshParts(component);
     }
 
+    private void OnMachineExaminableVerb(EntityUid uid, MachineComponent component, GetVerbsEvent<ExamineVerb> args)
+    {
+        if (!args.CanInteract || !args.CanAccess)
+            return;
+
+        var markup = new FormattedMessage();
+        this.EntityManager.EventBus.RaiseLocalEvent(uid, new UpgradeExamineEvent(ref markup));
```
Comment (EmoGarbage404) [inline Content.Server/Construction/ConstructionSystem.Machine.cs:40]:
```suggestion
        RaiseLocalEvent(uid, new UpgradeExamineEvent(ref markup));
```

**Review by EmoGarbage404 — APPROVED** [review]
Just needs the localization strings. Looks very cool.

**Comment by EmoGarbage404** [comment]
also just gonna say related to #11530 just so that it appears on the issue.

**Review by EmoGarbage404 — APPROVED** [review]
all the localization strings for the different upgrades should probably go in the `machine.ftl` file. I think it's better that way just so that they can all easily be found and tweaked if needed. It also reduces the chance that people accidentally duplicate the same string.

**Comment by 0x6273** [comment]
> all the localization strings for the different upgrades should probably go in the `machine.ftl` file. I think it's better that way just so that they can all easily be found and tweaked if needed. It also reduces the chance that people accidentally duplicate the same string.

But then if you want to find all loc strings relating to a single machine, you have 2 .ftl files you need to look in. And I think merge conflicts are a lot more likely if everything is in `machine.ftl`.

**Comment by 0x6273** [comment]
Rebased to add upgrade examining for the 3 machine upgrade PRs that just got merged.

**Comment by PJB3005** [comment]
Closed for crimes against humanity, and worse, Discord.

**Comment by 0x6273** [comment]
> Closed for crimes against humanity, and worse, Discord.

I'm sowwy >w<

**Comment by PJB3005** [comment]
After contacting the funny department I have decided this joke has grown old and you are obligated to one (1) PR-reopening as compensation.

**Comment by EmoGarbage404** [comment]
the catgirl play always works

**Pull request merged**